endstate_correction/simulation/base.py

```diff
@@ -0,0 +1,139 @@
+import abc
+import logging
+import os
+from typing import List, Literal, Union
+
+from openmm import Integrator, LangevinIntegrator, OpenMMException, Platform
+from openmm import unit
+from openmm.app import (
+    AmberInpcrdFile,
+    AmberPrmtopFile,
+    CharmmPsfFile,
+    DCDReporter,
+    NoCutoff,
+    PDBFile,
+    PME,
+    Simulation,
+    Topology,
+)
+from openmmml import MLPotential
+import numpy as np
+from ..constant import check_implementation
+from ..protocol import BSSProtocol
+from ..topology import AMBERTopology, CHARMMTopology
+
+
+class EndstateCorrectionBase(abc.ABC):
+    def __init__(
+        self,
+        top: Union[AMBERTopology, CHARMMTopology],
+        env: Literal["complex", "waterbox", "vacuum"],
+        ml_atoms: List[int],
+        protocol: BSSProtocol,
+        name: str = "endstate_correction",
+        work_dir: str = "./",
+        potential: str = "ani2x",
+        interpolate: bool = True,
+    ):
+        self.logger = logging.getLogger("EndstateCorrectionBase")
+        self.top = top
+        self.protocol = protocol
+        self.env = env
+        self.name = name
+        self.work_dir = work_dir
+        self._mm_topology = None
+        self._initial_coordinates = None
+
+        mm_system = self._createSystem()
+        ml_potential = MLPotential(potential)
+        ml_system = ml_potential.createMixedSystem(
+            self.get_mm_topology().topology,
+            mm_system,
+            ml_atoms,
+            interpolate=interpolate,
+            implementation="nnpops",
+        )
+        integrator = self.get_integrator()
+        _, platform = check_implementation()
+        platform = Platform.getPlatformByName(platform)
+        self.simulation = Simulation(
+            self.get_mm_topology().topology, ml_system, integrator, platform=platform
+        )
+
+    def get_mm_topology(self) -> Topology:
+        if self._mm_topology is None:
+            self._mm_topology = self._get_mm_topology()
+        return self._mm_topology
+
+    def get_initial_coordinates(self) -> unit.Quantity:
+        if self._initial_coordinates is None:
+            self._initial_coordinates = self._get_initial_coordinates()
+        return self._initial_coordinates
```

Comment:
changed this to provide coordinates instead of a class instance that contained coordinates.
Should add units here explicitly.